### PR TITLE
http2 tests: Make 'network' and 'user' sides more explicit

### PR DESCRIPTION
**Pull request by raboof**
In Http2ServerSpec information flows in 4 directions:
frames from the network to the server, requests from the server
to the user code, responses from the user code to the server, and
frames from the server to the network.

That can get fairly hard to follow at times. I wonder if it helps
when we explicitly put the test methods in 'user' and 'network'
objects?

(if we like this we should also apply it to Http2ClientSpec of
course)

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 1]
```diff
@@ -73,11 +73,11 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
...
           expectedResponseHeaderBlock: ByteString
         ): Unit = {
-          sendHEADERS(streamId, endStream = true, endHeaders = true, requestHeaderBlock)
-          expectRequest() shouldBe expectedRequest
+          network.sendHEADERS(streamId, endStream = true, endHeaders = true, requestHeaderBlock)
+          user.expectRequest() shouldBe expectedRequest
 
-          responseOut.sendNext(response.addAttribute(Http2.streamId, streamId))
-          val headerPayload = expectHeaderBlock(streamId)
+          user.responseOut.sendNext(response.addAttribute(Http2.streamId, streamId))
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:79]: Is that the same as `user.emitResponse`? Maybe that can be changed then as well.

**Review by jrudolph — APPROVED** [review]
LGTM, great idea, bit more verbose but helps being explicit in this case.

**Review by ignasi35 — APPROVED** [review]
love it!

**Comment by jrudolph** [comment]
@ignasi35 please squash commits when merging if they are not meant to be conserved.
